load-testing/initial-load-testing.md

```diff
@@ -0,0 +1,89 @@
+
+## Running the tests
+
+I carried out the upload on a t3.medium EC2 instance with 2 CPU cores and 4GB RAM. Our first users will have a similar CPU with 8GB of RAM so the test machine is lower spec.
+
+I did have to make some changes to various timeouts otherwise none of these would have worked.
+* The akka request timeout on the API, which defaults to 20s, was switched off.
+* The akka server timeout on the API, which defaults to 60s, was switched off.
+* The load balancer idle timout, which was set at 60s, was increased to 10 minutes.
+
+| File Type    | Number of files | File Size | Upload Time<sup>1</sup> | Backend metadata processing time<sup>2</sup>            | Export Time        |
+|--------------|-----------------|-----------|-------------------------|---------------------------------------------------------|--------------------|
+| wav          | 1               | 3Gb       | 80s                     | 150s                                                    | 288s               |
+| bmp          | 2               | 2.8Gb     | 137s                    | 155s                                                    | 1092s              |
+| random bytes | 1               | 5Gb       | 120s                    | 216s                                                    | Failed<sup>3</sup> |
+| pdf          | 2000            | 1.9Kb     | ~600s                   | ~200 checks failed<sup>4</sup> after ~1200s<sup>5</sup> | n/a                |
+| jpg          | 10000           | 160b      | Failed<sup>6</sup>      | n/a                                                     | n/a                |
+
+
+<sup>1</sup>The upload time includes the time taken to upload the file and run the client side processing
+
+<sup>2</sup>The backend processing time is the total time the backend checks took to run so the time taken by the slowest of the checks.
+
+<sup>3</sup> The maximum size for a non multipart upload is 5Gb. Because this file was random bytes, gzip didn't compress it much so the consignment export size was over this limit.
+
+<sup>4</sup> The api was responding too slowly so the API client in the download files lambda timed out.
+
+<sup>5</sup> The majority of backend checks that did pass should have completed a lot faster because each check for an individual file should only take around 1 minute. This suggests that SQS is sending a lot of requests to each individual lambda which is increasing the time it takes.
+
+<sup>6</sup> The browser crashed while trying to send the client side processing results to the API. The response json from the client metadata call is around 50Mb so the browser is running out of memory. I tried this in Chrome and Firefox. I did get the upload to succeed by using a t3.2xlarge which has 8 cores and 32Gb of RAM but this is much larger than the expected machines the users will use though.
+
+## Problems found during the tests
+
+### Timeouts
+There are 4 timeouts that were preventing large consignments being processed.
```

Comment:
What were the bottleneck steps? i.e. for large files and large consignments, what part of the process would fail?

Reply:
Large files was ok because there's only one or two of them. For multiple files, all of the requests were slow but the slowest was the `addFiles` mutation. I've put a line in saying that.